frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -166,12 +179,82 @@ public void after(java.util.List<? extends VFileEvent> events) {
 
   private void subscribeToGitRepositoryFilesChanges() {
     Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
-    GitRepositoryChangeListener listener = repository -> queueRepositoryUpdateAndModelRefresh();
+    GitRepositoryChangeListener listener = repository -> {
+      trackCurrentBranchChange(repository);
+      queueRepositoryUpdateAndModelRefresh();
+    };
+
     val messageBusConnection = project.getMessageBus().connect();
     messageBusConnection.subscribe(topic, listener);
     Disposer.register(project, messageBusConnection);
   }
 
+  @IgnoreUIThreadUnsafeCalls
```

Comment:
Lemme introduce more discipline in how that's used :general-spurdo:...

Reply:
https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1297

Reply:
Coz now it's hard to figure out what specific call is ignored

Reply:
good point